src/main/java/com/mongodb/hibernate/internal/MongoChecks.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal;
+
+import static com.mongodb.hibernate.internal.MongoAssertions.assertNotNull;
+
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Util class for checking, for example, argument values.
+ *
+ * <p>This class is not part of the public API and may be removed or changed at any time.
+ */
+public final class MongoChecks {
+    private MongoChecks() {}
+
+    /**
+     * Checks that {@code value} is not {@code null}.
+     *
+     * @param <T> The type of {@code value}.
+     * @param name The parameter name.
+     * @param value The argument value.
+     * @return {@code value}.
+     * @throws NullPointerException If {@code value} is {@code null}.
+     */
+    public static <T> T notNull(final String name, @Nullable T value) {
+        assertNotNull(name);
+        if (value == null) {
+            throw new NullPointerException(name + " must not be null");
+        }
+        return value;
+    }
```

Comment:
So we have already a `MongoAssertion` internal class to assert at runtime whether some value is null; I think given the usage of `NullAway`, this new class is not needed for method argument nullness checking, so it seems to me its function is duplicated with `MongoAssertion` class? The exception types are different, so that is the reason we introduce this class?